ml-agents-envs/mlagents/envs/env_manager.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import List, Dict, NamedTuple, Optional
+from mlagents.envs import AllBrainInfo, BrainParameters, Policy, ActionInfo
+
+
+class StepInfo(NamedTuple):
+    last_all_brain_info: Optional[AllBrainInfo]
+    current_all_brain_info: AllBrainInfo
+    all_action_infos: Optional[Dict[str, ActionInfo]]
+
+
+class EnvManager(ABC):
+    def __init__(self):
+        self.policies: Dict[str, Policy] = {}
+
+    def set_policy(self, brain_name: str, policy: Policy):
+        self.policies[brain_name] = policy
+
+    @abstractmethod
+    def step(self) -> List[StepInfo]:
+        pass
+
+    @abstractmethod
+    def reset(self, config=None, train_mode=True) -> List[StepInfo]:
+        pass
+
+    @abstractmethod
+    def external_brains(self) -> Dict[str, BrainParameters]:
+        pass
+
+    @property
+    @abstractmethod
+    def reset_parameters(self) -> Dict[str, str]:
```

Comment:
This is probably a `Dict[str, Any]` right?

Reply:
Actually it looks like this is required to be a `Dict[string, float]` by the protobuf definition: https://github.com/Unity-Technologies/ml-agents/blob/master/protobuf-definitions/proto/mlagents/envs/communicator_objects/environment_parameters_proto.proto

I'll change it to reflect that.